main.js

```diff
@@ -113,7 +113,26 @@ let end_experiment = {
     }
 }
 
-function getTimeline(stimuli) {
+/**
+ * Randomize a table of stimuli
+ */
+function randomizeStimuli(table) {
+    let shuffled = uil.randomization.randomizeStimuli(
+        table,
+        max_same_type=MAX_SUCCEEDING_ITEMS_OF_TYPE
+    );
+    if (shuffled !== null)
+        table = shuffled;
+    else
+        console.error('Unable to shuffle stimuli according constraints.');
+
+    return table; // shuffled table if possible original otherwise
```

Comment:
I'm not sure returning the original list is a good solution here. I think it should break, with a stronger effect than a console error.

Reply:
Yes, I'm in a bit of doubt here, because getting a hard error means losing the participant. It is ment to be notified when developing the experiment, but randomization should succeed in practice. I was thinking what is worse, losing all data or going to the experiment with a unshuffled list. Or in the best case we would have to ask to reload the page.

Reply:
The experiment now throws an error, with a helpful message to the developer.